src/ir/access_path_selectors_test.cc

```diff
@@ -0,0 +1,140 @@
+#include "src/ir/access_path_selectors.h"
+
+#include <string>
+#include <vector>
+
+#include "absl/hash/hash_testing.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_split.h"
+#include "src/common/testing/gtest.h"
+#include "src/ir/field_selector.h"
+#include "src/ir/selector.h"
+
+namespace raksha::ir {
+
+// Selector Access paths are easier to read and write in their string form, so
+// this is very useful for creating readable test inputs. This could possibly
+// be useful as a method in AccessPathSelectors itself, but, it's a bit
+// simplified right now, as it only splits on '.', the field separator
+// character. Also, I haven't found a use for it in non-test code yet.
+static absl::StatusOr<AccessPathSelectors> MakeSelectorAccessPathFromString(
+    const std::string str) {
+  const std::vector<std::string> selector_strs =
+      absl::StrSplit(str, '.', absl::SkipEmpty());
+
+  if (selector_strs.empty()) {
+    return absl::InvalidArgumentError(
+        "Expected a valid AccessPathSelectors string to have a"
+        "non-trivial leaf element.");
+  }
+
+  auto selector_rev_iter = selector_strs.rbegin();
+
+  // Create a leaf selector from the first selector string.
+  AccessPathSelectors selector_access_path =
+      AccessPathSelectors(Selector(FieldSelector(*selector_rev_iter)));
+  ++selector_rev_iter;
+
+  // Add all others as parents.
+  for (; selector_rev_iter < selector_strs.rend(); ++selector_rev_iter) {
+    selector_access_path = AccessPathSelectors(
+        Selector(FieldSelector(*selector_rev_iter)),
+        std::move(selector_access_path));
+  }
+
+  return selector_access_path;
+}
+
+class AccessPathEqualsTest :
+   public ::testing::TestWithParam<::std::tuple<std::string, std::string>> {};
+
+TEST_P(AccessPathEqualsTest, AccessPathsForEqualStringsCompareEquals) {
+  std::string access_path_str1;
+  std::string access_path_str2;
+  std::tie(access_path_str1, access_path_str2) = GetParam();
+
+  const bool access_paths_strs_equal = access_path_str1 == access_path_str2;
+
+  const absl::StatusOr<AccessPathSelectors> access_path1 =
+      MakeSelectorAccessPathFromString(access_path_str1);
+  ASSERT_TRUE(access_path1.ok());
+
+  const absl::StatusOr<AccessPathSelectors> access_path2 =
+      MakeSelectorAccessPathFromString(access_path_str2);
+  ASSERT_TRUE(access_path2.ok());
+
+  ASSERT_EQ(*access_path1 == *access_path2, access_paths_strs_equal);
```

Comment:
Just to make sure I understand, are you checking here that the comparison on constructed access paths is the same as the strings?

Reply:
Yes. I'm checking that the AccessPaths that I generate from these strings compare equal only when the original strings compare equal. I've added a comment describing this above the test body.